### PR TITLE
[1] Initialised FireStore and storing user data in DB

### PR DESCRIPTION
**Pull request by ankurnarkhede**
- Initialised firestore DB
- Storing user data on signup/login

### CODE REVIEW

File: controllers/authController.js [hunk 1]
```diff
@@ -1,26 +1,43 @@
...
+
+      userData = {
+        username: user.username,
+        name: user.displayName,
+        tokens: {
+          githubAccessToken: accessToken
+        }
+      }
+
+      const { newUser } = await users.addOrUpdate(userData)
```
[the remark this answers is not in the document]
Comment (ankurnarkhede) [reply controllers/authController.js]: Modified.

File: controllers/authController.js [hunk 2]
```diff
@@ -1,26 +1,43 @@
...
-  })(req, res, next)
+  try {
+    passport.authenticate('github', { session: false }, async (err, accessToken, user) => {
+      if (err) {
+        logger.error(err)
+        return res.boom.unauthorized('User cannot be authenticated')
+      }
+
+      userData = {
+        username: user.username,
```
[the remark this answers is not in the document]
Comment (ankurnarkhede) [reply controllers/authController.js]: Modified.

File: utils/firestore.js [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
+  keyFilename: './firestore-private-key.json'
+})
+
+const users = db.collection('users')
 
-module.exports = getFirestoreDB
+module.exports = {
+  db,
+  users
+}
```
[the remark this answers is not in the document]
Comment (ankurnarkhede) [reply utils/firestore.js]: Modified.

File: controllers/authController.js [hunk 4]
```diff
@@ -1,26 +1,43 @@
...
+        name: user.displayName,
```
[the remark this answers is not in the document]
Comment (ankurnarkhede) [reply controllers/authController.js]: Modified.
Also, I think we should stick to either camelCase or snake_case throughout the repos.
Will look over modifying this later.

File: utils/firestore.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ankurnarkhede) [reply utils/firestore.js]: Modified.

File: models/users.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * @param userData { Object }: User data object to be stored in DB
+ * @return {Promise<{newUser: boolean, userId: string}|{newUser: boolean, userId: string}>}
+ */
+const addOrUpdate = async (userData) => {
+  let userInfo
+
+  try {
+    // check if user already exists
+    const user = await userModel.where('username', '==', userData.username).limit(1).get()
```
[the remark this answers is not in the document]
Comment (ankurnarkhede) [reply models/users.js]: Modified.

**Review by ankushdharkar — APPROVED** [review]

**Pull request merged**